homework_weekend5/src/ba/bitcamp/homework_weekend5/task3and4/Protagonist.java

```diff
@@ -0,0 +1,168 @@
+package ba.bitcamp.homework_weekend5.task3and4;
+
+import ba.bitcamp.homework_weekend5.task1.StoryCharacter;
+import ba.bitcamp.homework_weekend5.task2.Goal;
+
+/**
+ * Describes a good guy in the story
+ * @author ajla
+ *
+ */
+public class Protagonist extends StoryCharacter {
+	
+	// Declaration and initialization of global variables
+	private static final int MAX_ALLIES = 3;
+	private static int ALLIES_NO = 0;
+	private static final int MIN_CHANCE_TO_SUCCEED = 50;
+	
+	// class properties
+	private Goal goal;
+	private StoryCharacter[] allies;
+	private StoryCharacter enemy;
+	
+	/**
+	 * Constructor
+	 * demands proper initialization of object
+	 * @param name
+	 * @param sex
+	 * @param isCapable
+	 * @param goal
+	 */
+	public Protagonist(String name, boolean sex, boolean isCapable, Goal goal) {
+		super(name, sex, isCapable);
+		this.goal = goal;
+		this.enemy = null;
+		this.allies = new StoryCharacter[MAX_ALLIES];
+	}
+
+	/**
+	 * @return the goal
+	 */
+	public Goal getGoal() {
+		return goal;
+	}
+
+	/**
+	 * @param goal the goal to set
+	 */
+	public void setGoal(Goal goal) {
+		this.goal = goal;
+	}
+
+	/**
+	 * @return the allies
+	 */
+	public StoryCharacter[] getAllies() {
+		return allies;
+	}
+
+	/**
+	 * @return the enemy
+	 */
+	public StoryCharacter getEnemy() {
+		return enemy;
+	}
+
+	/**
+	 * @param enemy the enemy to set
+	 */
+	public void setEnemy(StoryCharacter enemy) {
+		this.enemy = enemy;
+	}
+
+	/**
+	 * Adds an ally to the allies array
+	 * One protagonist cannot have more then 3 allies
+	 * @param ally
+	 */
+	public void addAlly(StoryCharacter ally) {	
+			if(ALLIES_NO == MAX_ALLIES) {
+				throw new ArrayIndexOutOfBoundsException("No more allies allowed!");
+			} else {
+				ALLIES_NO++;
+				for (int i = 0; i < allies.length; i++) {
+					if (allies[i] == null) {
+						allies[i] = ally;
+						break;
+					}
+				}
+			}
+	}
+	
+	/**
+	 * Calculate what chances the protagonist has 
+	 * to attain theirs goal. The calculation depends on
+	 * how many allies the protagonist has, and on their capabilities
+	 * also, it depends on goal difficulty
+	 */
+	public void attainGoal() {
```

Comment:
Same as other methods that can fail, printing failure message has no value to consumer of this method. Boolean should be returned that will be interpreted as an action result.